pyciemss/ouu/ouu.py

```diff
@@ -186,6 +186,7 @@ def __init__(
         maxfeval: int = 100,
         maxiter: int = 100,
         u_bounds: np.ndarray = np.atleast_2d([[0], [1]]),
+        progress_hook: Callable[[NDArray], None] = lambda x: None,  # update_progress
```

Comment:
Could we make this a `Callable[[torch.Tensor], None]` instead? I'm fine with numpy arrays being used internally to make use of scipy optimize, but I'd like to avoid numpy arrays leaking into any of the inputs for interfaces.

Reply:
Yes